create_lmodrc.py

```diff
@@ -17,6 +17,113 @@
 }
 """
 
+GPU_LMOD_RC ="""require("strict")
+local hook = require("Hook")
+local open = io.open
+
+local function read_file(path)
+    local file = open(path, "rb") -- r read mode and b binary mode
+    if not file then return nil end
+    local content = file:read "*a" -- *a or *all reads the whole file
+    file:close()
+    return content
+end
+
+-- from https://stackoverflow.com/a/40195356
+--- Check if a file or directory exists in this path
+function exists(file)
+    local ok, err, code = os.rename(file, file)
+    if not ok then
+        if code == 13 then
+            -- Permission denied, but it exists
+            return true
+        end
+    end
+    return ok, err
+end
+
+local function visible_hook(modT)
```

Comment:
I think we should describe what this hook is meant to do. As far as I can tell, the description should be something like:
"This hook checks if the directory $EESSI_CVMFS_REPO/host_injections/nvidia exists. If set up correctly, this directory contains the symlinks to the CUDA drivers on the host machine. This setup is required to make CUDA software from the EESSI stack find the drivers, they cannot run otherwise. If this directory does not exist, the symlinks are not setup, and using CUDA-enabled EESSI modules is not supported on the system. Hence, we hide those modules with this hook, to prevent users from loading modules that cannot work on their machine."

Reply:
Given we've added logic to help people know when they can and can't use a module, this bit is useless. Kicking it out